mlflow/sklearn/utils.py

```diff
@@ -157,8 +157,9 @@ def _get_classifier_metrics(fitted_estimator, prefix, X, y_true, sample_weight):
     https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html
     (3) f1_score:
     https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html
-    By default, we choose the parameter `labels` to be `None`, `pos_label` to be `1`,
-    `average` to be `weighted` to compute the weighted precision score.
+    By default, when `pos_label` is not specified (passed in as `None`), we set `average`
+    to `weighted` to compute the weighted score of theese metrics.
+    When the `pos_label` is specified (not `None`), we set `average` to `binary`.
```

Comment:
What will happen if the `pos_label` is set while it's actually a multi-label classification problem? Based on my understanding, it can throw something like `ValueError: Target is multiclass but average='binary'` from sklearn. Is this expected?
I think it's ok as it doesn't change current behavior. Do we want to make it clearer in `eval_and_log_metrics` docstring that user should only set `pos_label` for binary classification problem?

Reply:
Good point. I updated the documentation in eval_and_log_metrics to make this clear.